builder/amazon/common/ami_config.go

```diff
@@ -54,18 +64,46 @@ func (c *AMIConfig) Prepare(ctx *interpolate.Context) []error {
 				}
 			}
 
+			// Make sure that if we have region_kms_key_ids defined the regions in ami_regions are also in region_kms_key_ids
+			if len(c.AMIRegionKmsKeyIds) > 0 {
+				regionsInKeyMap := make([]string, 0, len(c.AMIRegionKmsKeyIds))
+				for reg := range c.AMIRegionKmsKeyIds {
+					regionsInKeyMap = append(regionsInKeyMap, reg)
+				}
+				if regionsMatch := stringInSlice(region, regionsInKeyMap); !regionsMatch {
+					errs = append(errs, fmt.Errorf("Region %s is in ami_regions but not in region_kms_key_ids", region))
+				}
+			}
+
 			regions = append(regions, region)
 		}
 
 		c.AMIRegions = regions
 	}
+	// Make sure that if we have region_kms_key_ids defined the regions in region_kms_key_ids are also in ami_regions
+	if len(c.AMIRegionKmsKeyIds) > 0 {
+		for kmsKeyRegion := range c.AMIRegionKmsKeyIds {
+			if regionsMatch := stringInSlice(kmsKeyRegion, c.AMIRegions); !regionsMatch {
```

Comment:
how about `if !stringInSlice(kmsKeyRegion, c.AMIRegions) {`?

I also think you can probably inline `stringInSlice` if it's not going to be used above